src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -143,12 +172,25 @@ static string BuildUrl(string url, params string?[] queryValues)
         //       request headers into consideration and generate the base URL accordingly.
         //       More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
         //       One potential workaround is to set ASPNETCORE_FORWARDEDHEADERS_ENABLED to true.
-        string baseUrl = request.Url.GetLeftPart(UriPartial.Authority);
+        string? baseUrl = (request != null) ? request.Url.GetLeftPart(UriPartial.Authority) : GetBaseUrl(client);
+        bool isFromRequest = request != null;
+        
+        if (baseUrl == null)
+        {
+            throw new InvalidOperationException("Base URL is null. Either use Functions bindings or provide an HTTP request to create the HttpPayload.");
```

Comment:
The BaseUrl could be null when 1. the http request provided is not valid; or 2. the DurableTaskClient is not `FunctionsDurableTaskClient` from which we get the host url information.